client/hotrod-client/src/main/java/org/infinispan/client/hotrod/ProtocolVersion.java

```diff
@@ -92,4 +92,31 @@ public static ProtocolVersion getBestVersion(int version) {
       }
       throw new IllegalArgumentException("Illegal version " + version);
    }
+
+   public static Codec getWithoutTracing() {
+      // We skip the last version (auto)
+      for (int i = VERSIONS.length - 2; i >= 0; i--) {
+         if (!VERSIONS[i].codec.supportsTracing())
+            return VERSIONS[i].codec;
+      }
+      throw new IllegalStateException("Codec without tracing not available!");
+   }
```

Comment:
can be replaced with
`public static final ProtocolVersion NO_TRACING_VERSION = PROTOCOL_VERSION_31;`

Reply:
+1 from me, since we can avoid the loop, maybe the name could be something like: `SAFE_HANDSHAKE_PROTOCOL_VERSION`. I would avoid using the word `tracing`, since the change on Hot Rod codec has a wider scope.

Reply:
I like `SAFE_HANDSHAKE_PROTOCOL_VERSION`